pydis_core/utils/paste_service.py

```diff
@@ -63,6 +623,18 @@ async def send_to_paste_service(
     if max_size > MAX_PASTE_SIZE:
         raise ValueError(f"`max_length` must not be greater than {MAX_PASTE_SIZE}")
 
+    if lexer not in SUPPORTED_LEXERS:
+        if paste_url == DEFAULT_PASTEBIN:
+            raise PasteUnsupportedLexerError(f"Lexer '{lexer}' not supported by pastebin.")
+
+        try:
+            async with http_session.get(f"{paste_url}/api/v1/lexer") as response:
+                response_json = await response.json()  # Supported lexers are the keys.
+        except Exception:
+            raise PasteUploadError("Could not fetch supported lexers from selected paste_url.")
+        if lexer not in response_json:
+            raise PasteUnsupportedLexerError(f"Lexer '{lexer}' not supported by pastebin.")
+
     contents_size = len(contents.encode())
     if contents_size > max_size:
         log.info("Contents too large to send to paste service.")
```

Comment:
Not related to this PR, but given the discussion about actual size including syntax highlighting, isn't this logic now not sufficient to catch all pastes that are too large?